heat/core/statistics.py

```diff
@@ -1100,6 +1132,231 @@ def mpi_argmin(a, b, _):
+            axis_slice = axis_slice[:axis] + (ceil_indices.tolist(),) + axis_slice[axis + 1 :]
+            if ceil_indices.max().item() == data.lshape[axis]:
+                if axis == data.split:
+                    highs = data.array_with_halos[axis_slice]
+                else:
+                    # max percentile is 100.0
+                    ceil_indices[ceil_indices.argmax()] -= 1
+                    axis_slice = (
+                        axis_slice[:axis] + (ceil_indices.tolist(),) + axis_slice[axis + 1 :]
+                    )
+                    highs = data._DNDarray__array[axis_slice]
+            else:
+                highs = data._DNDarray__array[axis_slice]
+            # from here on, local (torch) operations
+            weights_shape = data.numdims * (1,)
+            weights_shape = weights_shape[:axis] + (indices.shape[0],) + weights_shape[axis + 1 :]
+            weights = torch.sub(
+                indices.reshape(weights_shape), torch.floor(indices).reshape(weights_shape)
+            )
+
+            percentile = lows + weights * (torch.sub(highs, lows))
+
+            if axis != 0:
+                # permute to have the number of percentiles at dimension 0
+                dims = tuple(range(percentile.ndim))
+                permute_dims = (axis,) + dims[:axis] + dims[axis + 1 :]
+                percentile = percentile.permute(permute_dims)
+
+        return percentile
+
+    # sanitize input
+    if not isinstance(x, dndarray.DNDarray):
+        raise TypeError("expected x to be a ht.DNDarray, but was {}".format(type(x)))
+    if isinstance(axis, list) or isinstance(axis, tuple):
+        raise NotImplementedError("ht.percentile(), tuple axis not implemented yet")
+
+    if isinstance(q, dndarray.DNDarray):
+        if x.comm.is_distributed() and q.split is not None:
+            # q needs to be local
+            q.resplit_(axis=None)
+        q = q._DNDarray__array
+    elif isinstance(q, list) or isinstance(q, tuple):
+        q = torch.tensor(
+            q,
+            dtype=torch.promote_types(type(q[0]), x._DNDarray__array.dtype),
+            device=x._DNDarray__array.device,
+        )
+    elif np.isscalar(q) or isinstance(q, torch.tensor) and q.ndim == 0:
+        q = torch.tensor(
+            [q],
+            dtype=torch.promote_types(type(q), x._DNDarray__array.dtype),
+            device=x._DNDarray__array.device,
+        )
+    elif isinstance(q, torch.tensor):
+        pass
+    else:
+        raise TypeError(
+            "ht.tensor, torch.tensor, list or tuple supported, but q was {}".format(type(q))
+        )
+    nperc = q.numel()
+
+    # edge-case: x is a scalar. Return x
+    if x.numdims == 0:
+        percentile = x._DNDarray__array * q / q
+        return factories.array(percentile, split=None, dtype=x.dtype, device=x.device, comm=x.comm)
+
+    # no axis
+    if axis is None:
+        if x.numdims > 1:
+            x = x.copy()
```

Comment:
Puh, this could easily run into memory problems for large tensors, no?

Reply:
YES.

Reply:
Especially since you later copy x into data for sorting.... that I get, there is no way around it. But considering this you would make 2 extra copies..... Maybe think about fusing this for the axis=None case?